plugins/filter_aws/aws.c

```diff
@@ -182,191 +203,273 @@ static int cb_aws_init(struct flb_filter_instance *f_ins,
     return 0;
 }
 
-/* Get an IMDSv2 token */
-static int get_ec2_token(struct flb_filter_aws *ctx)
+/* Get VPC metadata, it called IMDS twice.
+ * First is for getting the Mac ID and combine into the path for VPC.
+ * Second call is using the VPC path to get the VPC id
+ *
+ * Note: it duplicates src/aws/flb_aws_imds.c 'flb_aws_imds_get_vpc_id' function,
+ * because it also needs to save 'vpc_id_len' in the ctx
+ */
+static int get_vpc_metadata(struct flb_filter_aws *ctx)
```

Comment:
@mwarzynski You do not need to duplicate this function. The function returns an `flb_sds_t` string. You can get the len of that string with `flb_sds_len(str)`.

Reply:
Ouh, yes! That's a great suggestion. I will change this today!